### PR TITLE
feat(TUP-25120)Merge nexus proxy setup

### PR DESCRIPTION
**Pull request by hzhao-talendbj**

### CODE REVIEW

File: main/plugins/org.talend.core.ui/src/main/java/org/talend/core/ui/services/IHadoopUiService.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+
+/**
+ * DOC hzhao  class global comment. Detailled comment
+ */
+public interface IHadoopUiService extends IService {
+
+    IPreferenceForm createDynamicDistributionPrefForm(Composite parent, PreferencePage prefPage);
+
+    static IHadoopUiService getInstance() {
```
Comment (wchen-talend) [inline main/plugins/org.talend.core.ui/src/main/java/org/talend/core/ui/services/IHadoopUiService.java]: why need a instance to get service here ?

File: main/plugins/org.talend.designer.maven.aether/src/main/java/org/talend/designer/maven/aether/util/DynamicDistributionAetherUtils.java [hunk 2]
```diff
@@ -368,6 +369,91 @@ public static String getHighestVersion(String remoteUrl, String username, String
...
+    public static TypedReturnCode<VersionRangeResult> checkConnection(String remoteUrl, String username, String password, String groupId,
+            String artifactId, String baseVersion, String topVersion, IDynamicMonitor monitor) throws Exception {
+        TypedReturnCode<VersionRangeResult> tc = new TypedReturnCode<VersionRangeResult>();
+        // maybe need a compatible limit using baseVersion and topVersion
+        if (monitor == null) {
+            monitor = new DummyDynamicMonitor();
+        }
+        //temp path
+        File tempDirectory = Files.createTempDirectory("nexusTemp_").toFile();
+        tempDirectory.deleteOnExit();
```
Comment (wchen-talend) [inline main/plugins/org.talend.designer.maven.aether/src/main/java/org/talend/designer/maven/aether/util/DynamicDistributionAetherUtils.java]: There will always be a temp folder after check connction
Better remove the temp folder after use it .

File: main/plugins/org.talend.core.ui/src/main/java/org/talend/core/ui/services/IHadoopUiService.java (hunk 1 above)
Comment (hzhao-talendbj) [reply main/plugins/org.talend.core.ui/src/main/java/org/talend/core/ui/services/IHadoopUiService.java]: Removed this getInstance and  if need to check, use serviceRegister.isServiceRegistered(IHadoopUiService.class)  instead

File: main/plugins/org.talend.designer.maven.aether/src/main/java/org/talend/designer/maven/aether/util/DynamicDistributionAetherUtils.java (hunk 2 above)
Comment (hzhao-talendbj) [reply main/plugins/org.talend.designer.maven.aether/src/main/java/org/talend/designer/maven/aether/util/DynamicDistributionAetherUtils.java]: Add a function recursiveDeleteOnShutdownHook,  will delete temp folder when shut down stduio

Comment (wchen-talend) [reply main/plugins/org.talend.designer.maven.aether/src/main/java/org/talend/designer/maven/aether/util/DynamicDistributionAetherUtils.java]: Why need a function when shut down studio ?
Can't delete the temp folder  in the finally block after check the connection status  ?

Comment (hzhao-talendbj) [reply main/plugins/org.talend.designer.maven.aether/src/main/java/org/talend/designer/maven/aether/util/DynamicDistributionAetherUtils.java]: Use FilesUtils.deleteFile in finally instead

**Pull request merged**